docs/developing_iree/profiling_with_tracy.md

```diff
@@ -0,0 +1,207 @@
+# Profiling with Tracy
+
+[Tracy](https://github.com/wolfpld/tracy) is a profiler that puts together in a
+single view:
+* Both CPU and GPU profiling.
+* Both sampling and instrumentation.
+* Both specifics of our own process, and whole-system profiling a la "systrace".
+
+Since Tracy relies on instrumentation, it requires IREE binaries to be built
+with a special flag to enable it.
+
+There are two components to Tracy. They communicate over a TCP socket.
+* The "client" is the program being profiled.
+* The "server" is the Tracy profiler UI.
+
+## The Tracy manual
+
+The primary source of Tracy documentation, including for build instructions, is
+a PDF manual that's part of each numbered release. To find the latest one,
+navigate [here](https://github.com/wolfpld/tracy/releases) and search for
+`tracy.pdf`.
+
+## Building the Tracy UI (the "server")
+
+This is explained in section 2.3 of the manual for Windows and Linux. Here we
+give some more detailed instructions for some systems.
+
+The IREE repository contains its own clone of the Tracy repository in
+`third_party/tracy`, so there is no need to make a separate clone of it. You can
+use one if you want, but be aware that the Tracy client/server protocol gets
+updated sometimes (it's at version 43 now!). Building both sides from the same
```

Comment:
"it's at version 43 now!" will almost immediately be outdated 
can keep if you want :P

Reply:
done